### PR TITLE
Documentation additions about code style and philosophy

### PR DESCRIPTION
**Pull request by joshrabinowitz**
Changes to `.github/PULL_REQUEST_TEMPLATE.md` and `CONTRIBUTING.md`

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 <!-- Thanks for sending a pull request!
 
 Here's how it's done:
-0. If you are planning a large feature, please, discuss it first in a separate issue
+0. If you are planning a large feature, please, discuss it first in a separate issue.
+   See also CONTRIBUTING.md if you haven't already.
```
Comment (sobolevn) [inline .github/PULL_REQUEST_TEMPLATE.md]: A link maybe?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 1. Make sure that you open your pull request against the `master` branch
-2. Make sure that tests pass
-3. Make sure that your code has the same style
+2. Make sure that your code has the same style as the surrounding code and git-secret in general
```
Comment (sobolevn) [inline .github/PULL_REQUEST_TEMPLATE.md]: We should mention `shellcheck` here

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -37,6 +37,46 @@ Only required if dealing with manuals, `gh-pages` or releases:
...
+   This also aids future development and helps minimize bugs.
+
+4. consistent: Write code that is consistent with the surrounding code and the rest of the git-secret code base.
+   Every code base has its own conventions and style that develop and accrete over time.
+
+   Consistency also means that the inputs and outputs of git-secret should be as consistent as reasonable
+   with related unix and git tools, and follow the 'rule of least surprise', 
+   also known as the 'principle of least astonishment': <https://en.wikipedia.org/wiki/Principle_of_least_astonishment>
+
+We wrote this to clarify our thinking about how git-secret should be written.  Of course, these are philosophical goals,
```
Comment (sobolevn) [inline CONTRIBUTING.md]: Love this whole part ❤️

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -37,6 +37,46 @@ Only required if dealing with manuals, `gh-pages` or releases:
...
+   Consistency also means that the inputs and outputs of git-secret should be as consistent as reasonable
+   with related unix and git tools, and follow the 'rule of least surprise', 
+   also known as the 'principle of least astonishment': <https://en.wikipedia.org/wiki/Principle_of_least_astonishment>
+
+We wrote this to clarify our thinking about how git-secret should be written.  Of course, these are philosophical goals, 
+not necessities for releasing code, so balancing these four ideals _perfectly_ is both unwarranted and impossible.
+
+### Writing PRs
+
+If you're planning a large change to git-secret (for example, a lot of lines/characters of diffs, affecting multiple commands,
```
Comment (sobolevn) [inline CONTRIBUTING.md]: It should be `git-secret` for consistency

File: CONTRIBUTING.md (hunk 3 above)
Comment (joshrabinowitz) [reply CONTRIBUTING.md]: Thank you @sobolevn !!!

File: CONTRIBUTING.md (hunk 4 above)
Comment (joshrabinowitz) [reply CONTRIBUTING.md]: I don't understand, isn't that what it says?

Comment (joshrabinowitz) [reply CONTRIBUTING.md]: oh I see, with backticks

**Pull request merged**